spec/requests/api/v1/merchants_requests_spec.rb

```diff
@@ -122,25 +122,28 @@
       expect(attributes[:name]).to be_a(String)
     end
 
-    it 'can delete a merchant' do
-      merchant = Merchant.create!(name: "Brown and Sons")
-      item1 = merchant.items.create!(name: "Item 1" , unit_price: 20)
-      item2 = merchant.items.create!(name: "Item 2", unit_price: 30)
-      customer = Customer.create!(first_name: "John", last_name: "Doe")
-      invoice1 = Invoice.create!(merchant: merchant, customer: customer, status: "pending")
-      invoice2 = Invoice.create!(merchant: merchant, customer: customer, status: "completed")
-
-      expect(Merchant.all).to include(merchant)
-      expect(merchant.items).to include(item1, item2)
-      expect(merchant.invoices).to include(invoice1, invoice2)
+    it 'can delete a merchant and associated records, returns 204 no content' do
+      item1 = @merchant_1.items.create!(name: "Item 1", unit_price: 20)
```

Comment:
I love that you refactored your tests to use the new before each block. So dynamic now!